src/flags.ts

```diff
@@ -55,3 +56,8 @@ export const isColorSupported =
 export const nodeVersion =
   (_process.versions?.node || "").replace(/^v/, "") || null;
 export const nodeMajorVersion = Number(nodeVersion?.split(".")[0]) || null;
+
+/** Detect the runtime type */
+export const isNode = detectRuntime() === "node";
+export const isDeno = detectRuntime() === "deno";
+export const isBun = detectRuntime() === "bun";
```

Comment:
For consistency, we might want to define `export const runtimeName = detectRuntime().name` (this also remove need of calling same function over again)

To reduce bundle size further (when we only want to check if `isDeno`, we can directly have something like `export const isDeno = globalThis.deno`. This way when tree-shaking, the other checks are not bundled.

Reply:
I'm not sure about `export const isDeno = globalThis.deno`, this would duplicate the runtime detection logic and might be forgotten in the future when updating a check.

Reply:
This duplicates our internal logic, yes. But imagine if a consumer library, only cares about if is running in Deno or not. When they import `isDeno` this way, they don't pull out the whole runtime detection logic. So their could can effectively be treeshaken. (also little bit of runtime overhead reduction).

I agree with you about maintenance btw. Happy to refactor and export from same `runtimes.ts` file?

Reply:
I've updated the code to make it lighter if you only need e.g `isBun`, but also avoid duplicating the same logic by reusing these flags. Let me know if that works for you!